src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3067 @@
+    BeginNormalPaletteFade(3, 1, 1, RGB_WHITE, data->fadeDuration, 1, HEAP_ID_REGISTER_HALL_OF_FAME);
+}
+
+static void RegisterHallOfFame_AfterFullParty_FadeToBlack(RegisterHallOfFameData *data) {
+    BeginNormalPaletteFade(0, 0, 0, RGB_BLACK, data->fadeDuration, 1, HEAP_ID_REGISTER_HALL_OF_FAME);
+}
+
+static void VBlankTask_RegisterHallOfFame_WholePartyCongrats(SysTask *task, void *taskData) {
+    RegisterHallOfFameData *data = (RegisterHallOfFameData *)taskData;
+    if (data->subprocCallback != NULL && !data->subprocCallback(data)) {
+        data->subprocCallback = NULL;
+    }
+    RegisterHallOfFame_G3Commit(data);
+    DoScheduledBgGpuUpdates(data->bgConfig);
+    sub_0200D020(data->spriteGfxHandler);
+    thunk_OamManager_ApplyAndResetBuffers();
+    OS_SetIrqCheckFlag(OS_IE_V_BLANK);
+}
+
+static void RegisterHallOfFame_WholePartyScene_InitBGs(RegisterHallOfFameData *data) {
+    {
+        GraphicsModes graphicsModes = sGraphicsModes_WholePartyScene;
+        SetBothScreensModesAndDisable(&graphicsModes);
+    }
+
+    {
+        BgTemplate bgTemplate = sBgTemplate_WholePartyScene_Main1;
+        InitBgFromTemplate(data->bgConfig, GF_BG_LYR_MAIN_1, &bgTemplate, GF_BG_TYPE_TEXT);
+        BgClearTilemapBufferAndCommit(data->bgConfig, GF_BG_LYR_MAIN_1);
+        BG_ClearCharDataRange(GF_BG_LYR_MAIN_1, 0x20, 0, HEAP_ID_REGISTER_HALL_OF_FAME);
+    }
+
+    {
+        BgTemplate bgTemplate = sBgTemplate_WholePartyScene_Main2;
+        InitBgFromTemplate(data->bgConfig, GF_BG_LYR_MAIN_2, &bgTemplate, GF_BG_TYPE_TEXT);
+    }
+
+    {
+        BgTemplate bgTemplate = sBgTemplate_WholePartyScene_Main3;
+        InitBgFromTemplate(data->bgConfig, GF_BG_LYR_MAIN_3, &bgTemplate, GF_BG_TYPE_TEXT);
+    }
+
+    GfGfx_EngineATogglePlanes(GX_PLANEMASK_BG1, GF_PLANE_TOGGLE_OFF);
+    GfGfx_EngineATogglePlanes(GX_PLANEMASK_BG2, GF_PLANE_TOGGLE_OFF);
+    GfGfx_EngineATogglePlanes(GX_PLANEMASK_BG3, GF_PLANE_TOGGLE_OFF);
+}
+
+static void RegisterHallOfFame_WholePartyScene_DeleteBgTilemapBuffers(RegisterHallOfFameData *data) {
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_3);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_2);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_1);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_0);
+}
+
+static void RegisterHallOfFame_WholePartyScene_LoadBgGfx(RegisterHallOfFameData *data) {
+    GfGfxLoader_LoadScrnDataFromOpenNarc(data->narcA094, 5, data->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, TRUE, HEAP_ID_REGISTER_HALL_OF_FAME);
+    GfGfxLoader_LoadScrnDataFromOpenNarc(data->narcA094, 6, data->bgConfig, GF_BG_LYR_MAIN_3, 0, 0, TRUE, HEAP_ID_REGISTER_HALL_OF_FAME);
+    GfGfxLoader_LoadCharDataFromOpenNarc(data->narcA094, 7, data->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, TRUE, HEAP_ID_REGISTER_HALL_OF_FAME);
+    GfGfxLoader_GXLoadPalFromOpenNarc(data->narcA094, 8, GF_PAL_LOCATION_MAIN_BG, (enum GFPalSlotOffset)0, 0x20, HEAP_ID_REGISTER_HALL_OF_FAME);
```

Comment:
GF_PAL_SLOT_0_OFFSET

Reply:
a future PR will elimitate the `GF_PAL_SLOT_XXX_OFFSET` enums and remove the type constraint

Reply:
I know, that's fair enough, however, I'd prefer if the constants are used rather than a cast like this while they still exist

it doesn't really matter and I'm not gonna block anything if it's still around

Reply:
disagree, the cast makes it easier to remove later on